### PR TITLE
coverage(PrizePool): increase coverage percentage

### PR DESCRIPTION
**Pull request by kamescg**
Added getters for public variables.

@asselstine  Should we update he public access for ticket, prizeStrategy, balance and liquidityCap, so they're only accessible through the `get` functions?

image

### CODE REVIEW

File: test/prize-pool/PrizePool.test.ts [hunk 1]
```diff
@@ -2,38 +2,39 @@ import { Signer } from '@ethersproject/abstract-signer';
...
 const { getContractFactory, getSigners, Wallet } = ethers;
 const { parseEther: toWei } = utils;
-
 const debug = require('debug')('ptv3:PrizePool.test');
-
 const NFT_TOKEN_ID = 1;
 
 describe('PrizePool', function () {
-  let contractsOwner: SignerWithAddress;
+  let wallet1: SignerWithAddress;
```
[the remark this answers is not in the document]
Comment (kamescg) [reply test/prize-pool/PrizePool.test.ts:16]: We use wallet1 in all the other tests.

File: contracts/prize-pool/PrizePool.sol [hunk 2]
```diff
@@ -83,7 +78,10 @@ abstract contract PrizePool is IPrizePool, Ownable, ReentrancyGuard, IERC721Rece
   function awardBalance() external override view returns (uint256) {
     return _currentAwardBalance;
   }
-
+  /// @inheritdoc IPrizePool
+  function accountedBalance() external override view returns (uint256) {
```
[the remark this answers is not in the document]
Comment (kamescg) [reply contracts/prize-pool/PrizePool.sol:82]: Wasn't sure which one we wanted to keep.

File: contracts/prize-pool/PrizePool.sol [hunk 3]
```diff
@@ -264,7 +278,7 @@ abstract contract PrizePool is IPrizePool, Ownable, ReentrancyGuard, IERC721Rece
   }
 
   /// @inheritdoc IERC721Receiver
-  function onERC721Received(address,address,uint256,bytes calldata) external pure override returns (bytes4) {
+  function onERC721Received(address,address,uint256,bytes calldata) public pure override returns (bytes4) {
```
[the remark this answers is not in the document]
Comment (kamescg) [reply contracts/prize-pool/PrizePool.sol:281]: Chuuu right @PierrickGT - removed public and just call directly.

File: contracts/interfaces/IPrizePool.sol [hunk 4]
```diff
@@ -119,17 +124,32 @@ interface IPrizePool {
   /// @return The underlying balance of assets
   function balance() external returns (uint256);
 
+  /**
+    * @notice Read internal Ticket accounted balance.
+  */
+  function getAccountedBalance() external view returns (uint256);
```
[the remark this answers is not in the document]
Comment (kamescg) [reply contracts/interfaces/IPrizePool.sol:130]: Affirmative.

**Comment by asselstine** [comment]
@kamescg 

> @asselstine Should we update he public access for ticket, prizeStrategy, balance and liquidityCap, so they're only accessible through the get functions?

Yes!

File: contracts/prize-pool/PrizePool.sol [hunk 5]
```diff
@@ -83,7 +78,10 @@ abstract contract PrizePool is IPrizePool, Ownable, ReentrancyGuard, IERC721Rece
   function awardBalance() external override view returns (uint256) {
     return _currentAwardBalance;
   }
-
+  // /// @inheritdoc IPrizePool
+  // function accountedBalance() external override view returns (uint256) {
+  //   return _ticketTotalSupply();
+  // }
```
Comment (PierrickGT) [inline contracts/prize-pool/PrizePool.sol:84]: Remove this comment.

File: contracts/test/PrizePoolHarness.sol [hunk 6]
```diff
@@ -55,4 +59,12 @@ contract PrizePoolHarness is PrizePool {
   function _redeem(uint256 redeemAmount) internal override returns (uint256) {
     return stubYieldSource.redeemToken(redeemAmount);
   }
-}
+
+  // function mockOnERC721Received(bytes calldata sig) external pure returns (bytes4) {
+  //   return super.onERC721Received(address(0), address(0), 0, sig);
+  // }
```
Comment (PierrickGT) [inline contracts/test/PrizePoolHarness.sol:65]: Remove this comment.

File: test/prize-pool/PrizePool.test.ts [hunk 7]
```diff
@@ -277,174 +388,172 @@ describe('PrizePool', function () {
...
     });
-  });
-});
+  })
+
+  /*============================================ */
+  // Internal Functions ------------------------
+  /*============================================ */
+  describe('Internal Functions', () => {
+    it('should get the curren block.timestamp', async () => {
```
Comment (PierrickGT) [inline test/prize-pool/PrizePool.test.ts:552]:
```suggestion
    it('should get the current block.timestamp', async () => {
```

File: test/prize-pool/PrizePool.test.ts [hunk 8]
```diff
@@ -277,174 +388,172 @@ describe('PrizePool', function () {
...
     });
-  });
-
-  describe('onERC721Received()', () => {
-    it('should receive an ERC721 token when using safeTransferFrom', async () => {
-      expect(await erc721Token.balanceOf(prizePool.address)).to.equal('0');
 
-      await depositNftIntoPrizePool(contractsOwner.address);
+    describe('onERC721Received()', () => {
+      it('should return the inteface selector', async () => {
```
Comment (PierrickGT) [inline test/prize-pool/PrizePool.test.ts:535]:
```suggestion
      it('should return the interface selector', async () => {
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

**Review by PierrickGT — APPROVED** [review]

**Pull request merged**